examples/vnet/doca_gw_dpdk.c

```diff
@@ -68,18 +99,190 @@ int doca_gw_port_stop(struct doca_gw_port *port)
     }
     return 0;
 }
+static void build_eth_flow_item(struct gw_item_param  *param,
+					struct doca_gw_match *match)
+{
+	struct rte_flow_item *item = param->item;
+
+	match = match;
+	item->type = RTE_FLOW_ITEM_TYPE_ETH;
+}
+
+#define GET_SRC_IP(match, type) ((type == OUTER_IP) ? match->out_src_ip : match->in_src_ip)
+#define GET_DST_IP(match, type) ((type == OUTER_IP) ? match->out_dst_ip : match->in_dst_ip)
+
+static int set_ipv4_flow_item(struct gw_item_param *param,
+									struct doca_gw_match *match)
+{
+	struct rte_flow_item_ipv4 *spec = &param->item_param.ip4.spec;
+	//struct rte_flow_item_ipv4 *mask = &param->item_param.ip4.mask;
+
+	if ((param->flags & SET_INNER_SRC_IP) && match->in_src_ip.a.ipv4_addr)
+		spec->hdr.src_addr = match->in_src_ip.a.ipv4_addr;
+	if ((param->flags & SET_OUTER_SRC_IP) && match->out_src_ip.a.ipv4_addr)
+		spec->hdr.src_addr = match->out_src_ip.a.ipv4_addr;
+	if ((param->flags & SET_INNER_DST_IP) && match->in_dst_ip.a.ipv4_addr)
+		spec->hdr.dst_addr = match->in_dst_ip.a.ipv4_addr;
+	if ((param->flags & SET_OUTER_DST_IP) && match->out_dst_ip.a.ipv4_addr)
+		spec->hdr.dst_addr = match->out_dst_ip.a.ipv4_addr;
+	return 1;
+}
+
+static void build_ipv4_flow_item(struct gw_item_param *param,
+						struct doca_gw_match *match, uint8_t type)
+{
+	struct rte_flow_item *item = param->item;
+	struct doca_ip_addr src_ip = GET_SRC_IP(match, type);
+	struct doca_ip_addr dst_ip = GET_DST_IP(match, type);
+	struct rte_flow_item_ipv4 *spec = &param->item_param.ip4.spec;
+	struct rte_flow_item_ipv4 *mask = &param->item_param.ip4.mask;
+
+	item->type = RTE_FLOW_ITEM_TYPE_IPV4;
+	if (src_ip.a.ipv4_addr){
+		spec->hdr.src_addr = src_ip.a.ipv4_addr;
+		if (src_ip.a.ipv4_addr == 0xffffff)
```

Comment:
what about the mask?

Reply:
Bill,  can you call the fields with some prefix , DOCA_GW_OUTER_IP_FLAG...etc. you can use DG_OUTER_IP_FLAG

Reply:
currently the match fields not include the mask part.
DOCA_OUTER_SRC_IP or DOCA_OUTER_SRC_IP_FLAG ?